lib/util/device-os-version.coffee

```diff
@@ -2,7 +2,7 @@ isEmpty = require('lodash/isEmpty')
 { isProvisioned } = require('.')
 
 exports.normalizeDeviceOsVersion = (device) ->
-	if isEmpty(device.os_version) and isProvisioned(device)
+	if device.os_version? and isEmpty(device.os_version) and isProvisioned(device)
```

Comment:
Would it make more sense to use `os_version in device`?

Reply:
I think I'd rather keep it this way, since it's a bit more explicit and the TS equivalent (`device.os_version != null`) is enough to let TS infer that `device.os_version` is `undefined | null` (and so catch errors better), which the alternative doesn't. Small thing, but I'd rather lean in that direction.

Reply:
So, my concern was that,
if you didn't `select` the `os_version`, but you `select` the `supervisor_version` & `last_connectivity_event` and the `isProvisioned` test passes, then you **will** get a `'Resin OS 1.0.0-pre'` `os_version` which might be incorrect.
On the other hand, if you didn't select it, then we don't expect that you will be using it.

Reply:
I don't think that's true. Because this is `and` all the way through, if you don't select `os_version` then `device.os_version?` is false, and so this does nothing.

This code only gives you the `-pre` version if _all_ the below are true:

* Your query did include an os_version result
* That result is an empty string
* The device is provisioned

I don't think there's any case where you do that, but it's not genuinely a `1.0.0-pre` device. Am I missing something?

(More generally: if you think there's a bug or incorrect behaviour like this, don't approve the PR! I'm very happy to discuss things like this before merge, it's much better than accidentally releasing a bug somewhere)